onnx/version_converter/adapters/softmax_12_13.h

```diff
@@ -25,57 +25,45 @@ class Softmax_12_13 final : public Adapter {
     if (old_axis == input_rank - 1)
       node->i_(kaxis, -1);
     else {
+      //    -- shape ------------------
+      //   /                           |
+      // ----- flatten -- softmax -- reshape
+
+      // get original softmax's input shape
+      Symbol kShape("Shape");
+      Node* shape = graph->create(kShape);
+      shape->addInput(node->inputs()[0]);
+      shape->insertBefore(node);
+
       // Insert Flatten node before softmax
       Node* flatten = graph->create(kFlatten);
       flatten->addInput(node->inputs()[0]);
       flatten->insertBefore(node);
       flatten->i_(kaxis, old_axis);
       node->replaceInput(0, flatten->output());
 
-      if (old_axis == 0) {
-        node->i_(kaxis, 1);
-      } else {
-        node->i_(kaxis, -1);
-      }
+      // Softmax along the last axis of the flattened 2D tensor
+      node->i_(kaxis, -1);
 
       // Insert Reshape node after softmax
       const std::string original_output_name = node->output()->uniqueName();
       const use_list original_uses(node->output()->uses());
       node->output()->setUniqueName(original_output_name + "_intermediate");
       Node* reshape = graph->create(kReshape);
       reshape->addInput(node->outputs()[0]);
+      reshape->addInput(shape->output());
       reshape->output()->setUniqueName(original_output_name);
       reshape->insertAfter(node);
 
-      // Set shape input of Reshape
-      const std::vector<Dimension>& target_shape = flatten->inputs()[0]->sizes();
-
-      ONNX_ASSERTM(
-          target_shape.size() != 0,
-          "Version conversion for Softmax failed because "
-          "input shape is unknown.");
-
-      Tensor t;
-      t.elem_type() = TensorProto_DataType_INT64;
-      t.sizes() = std::vector<int64_t>{static_cast<int64_t>(target_shape.size())};
-      auto& data = t.int64s();
-      for (Dimension dim : target_shape) {
-        data.emplace_back(dim.dim);
-      }
-      Node* constant = graph->create(kConstant);
-      constant->insertBefore(node);
-      constant->t_(kvalue, t);
-      reshape->addInput(constant->output());
-
       // Fix outputs & wiring
-      node->output()->wipeSizes();
-      reshape->output()->setSizes(target_shape);
+      reshape->output()->setSizes(node->output()->sizes());
```

Comment:
```suggestion
      if (node->output()->sizes().size() != 0) {
        reshape->output()->setSizes(node->output()->sizes());
      }
```
Potentially it will make the output rank of Reshape as zero if the original Softmax's output is unknown. Would it better not to assign it if the original rank is zero?

Reply:
I tested this condition with the failed ONNX Model models (ssd-12.onnx, MaskRCNN-12.onnx, gpt2-lm-head-bs-12.onnx) and they can pass both onnx.checker with full_check and onnxruntime.InferenceSession.

Reply:
Yes, this looks better. Thanks for the suggestion!